### ISSUE TITLE
Performance mismatch with trtexec

### ISSUE THREAD

**Issue by hx-byte** [NVIDIA/TensorRT-Edge-LLM#14]
### env
jetson thor, jetpack 7.1

### model
qwen3-vl-4b

### export(host)

1. `tensorrt-edgellm-quantize-llm --model_dir Qwen/Qwen3-VL-2B-Instruct --quantization fp8 --output_dir quantized/qwen3-vl-2b-fp8`
2. `tensorrt-edgellm-export-llm --model_dir quantized/qwen3-vl-2b-fp8 --output_dir onnx_models/qwen3-vl-2b-fp8`
3. `tensorrt-edgellm-export-visual --model_dir Qwen/Qwen3-VL-2B-Instruct --output_dir onnx_models/qwen3-vl-2b-fp8/visual_enc_onnx --quantization fp8`

### build(thor)

1. `./build/examples/llm/llm_build     --onnxDir /home/jt/zhx/ws/edge/onnx/qwen3-vl-2b-fp8 --engineDir /home/jt/zhx/ws/edge/engines/qwen3-vl-2b-fp8   --maxBatchSize 1 --maxInputLen 2048 --maxKVCacheCapacity 4096 --vlm --minImageTokens 128 --maxImageTokens 2048`
2. `./build/examples/multimodal/visual_build     --onnxDir /home/jt/zhx/ws/edge/onnx/qwen3-vl-2b-fp8/visual_enc_onnx     --engineDir /home/jt/zhx/ws/edge/visual_engines/qwen3-vl-2b-fp8-debug     --minImageTokens 128     --maxImageTokens 2048     --maxImageTokensPerImage 2048`

### run
1. input.json
`{
    "batch_size": 1,
    "temperature": 1.0,
    "top_p": 1.0,
    "top_k": 50,
    "max_generate_length": 128,
    "requests": [
        {
            "messages": [
                {
                    "role": "user",
                    "content": 
                    [
                        {
                            "type": "image",
                            "image": "https://qianwen-res.oss-cn-beijing.aliyuncs.com/Qwen-VL/assets/demo.jpeg"
                        },
                        {
                            "type": "text",
                            "text": "Describe this image."
                        }
                    ]
                }
            ]
        }
    ]
}`
2. llm_inference profile
`./build/examples/llm/llm_inference   --engineDir ../engines/qwen3-vl-2b-fp8     --multimodalEngineDir ../visual_engines/qwen3-vl-2b-fp8     --inputFile ../input_with
_image.json     --outputFile ../output_2b_fp8.json   --dumpProfile     --profileOutputFile profile_2b_fp8.json     --warmup 2`

<img width="1490" height="1806" alt="Image" src="https://github.com/user-attachments/assets/60a0f701-3313-43c5-99f8-9580741af736" />

3. trtexec profile
`/usr/src/tensorrt/bin/trtexec --loadEngine=/home/jt/zhx/ws/edge/visual_engines/qwen3-vl-2b-fp8/visual.engine --shapes=input:1920x1536,rotary_pos_emb:1920x32,attention_mask:1x1920x1920,fast_pos_embed_idx:4x1920,fast_pos_embed_weight:4x1920 --warmUp=10 --useManagedMemory --iterations=10`

<img width="3686" height="1484" alt="Image" src="https://github.com/user-attachments/assets/5c8b08a7-b0f6-4fc5-a2b2-52d18651b5fc" />

4. log
The multimodal_processing phase executed via llm_inference takes as long as 500 milliseconds, while trtexec only requires approximately 38 milliseconds for inputs of the same shape. I attempted to add the following code before and after line 775 at https://github.com/NVIDIA/TensorRT-Edge-LLM/blob/50a61d063c7d3076fd2eb0aa03d91023065b772c/cpp/multimodal/qwenViTRunner.cpp:
`        CUDA_CHECK(cudaStreamSynchronize(stream));
        auto startTime = std::chrono::high_resolution_clock::now();
        bool enqueueStatus = mContext->enqueueV3(stream);
        if (!enqueueStatus)
        {
            LOG_ERROR("QwenViTRunner::infer(): Failed to enqueue engine.");
            return false;
        }
        CUDA_CHECK(cudaStreamSynchronize(stream));
        auto endTime = std::chrono::high_resolution_clock::now();
        LOG_INFO("############# QwenViTRunner::infer(): Inference completed in %.3f ms.",
            std::chrono::duration<double, std::milli>(endTime - startTime).count());`
The logs still indicated a duration close to 500ms. Why is there such a significant discrepancy between this result and that of trtexec?

**Comment by hx-byte** [comment]
> > Thanks for your feedback! Our Qwen3-VL's engine input is HW instead of number of tokens. In Qwen3-VL, HW is 4x number of tokens, so your trtexec command it should instead be 7680 instead of 1920 to match the Edge-LLM implementation. Since MHA is O(n^2), it is possible that this 4x image token difference would bring up from 35 to 500 (~16x scaling). Could you measure again with 7680? Thanks!
> 
> Thank you very much for your solution！ The Qwen3-VL-4B visual encoder does perform a merge operation during output. With an input shape of 7980, the latency of trtexec is around 650 ms, which is slower than the ~500 ms latency of llm_inference. Do you know the underlying tricks behind this? Here are the execution parameters and results of my trtexec run： `/usr/src/tensorrt/bin/trtexec --loadEngine=/home/jt/zhx/ws/edge/visual_engines/qwen3-vl-2b-fp8/visual.engine --shapes=input:7920x1536,rotary_pos_emb:7920x32,attention_mask:1x7920x7920,fast_pos_embed_idx:4x7920,fast_pos_embed_weight:4x7920 --warmUp=10 --useManagedMemory --useCudaGraph --useSpinWait --iterations=10`
> 
> <img alt="Image" width="2000" height="1252" src="https://private-user-images.githubusercontent.com/243952603/535402286-ee18c9a5-ef40-4ca1-83ce-fddd33634e65.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjgzNzY3NjYsIm5iZiI6MTc2ODM3NjQ2NiwicGF0aCI6Ii8yNDM5NTI2MDMvNTM1NDAyMjg2LWVlMThjOWE1LWVmNDAtNGNhMS04M2NlLWZkZGQzMzYzNGU2NS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwMTE0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDExNFQwNzQxMDZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hMDg3M2ZmMzgwYTg4M2JhN2I1YTZjN2M3OThhNjI2M2JiOWViODg5NTFkNzRjMjIxMDg1Njg4MjQ4OGE4ZDFhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.pSCPS_WzKyULhHqElXJi6duG-bizevK-JlXG4oACToQ">

I found that this was entirely due to my use of the --useManagedMemory parameter. After removing this parameter, the time consumption is basically consistent with that of trtexec. Thank you very much!
